common/src/hand_off.rs

```diff
@@ -120,9 +121,21 @@ impl Default for FirmwareHandoffTable {
 
 impl FirmwareHandoffTable {
     /// Perform valdity check of the table's data.
+    /// The fields below should have been populated by ROM with
+    /// valid data before it transfers control to mutable code.
     pub fn is_valid(&self) -> bool {
```

Comment:
should rename to is_valid_for_fmc(), since the FMC requirements are ot the same as RT requirements.

Then create new routine, is_valid_for_rt() than covers what is needed there.

Reply:
As of now , the is_valid refers to values populated by ROM before it exits as per  comments , but we can extend its functionality.

Reply:
As per conversation, with @FerralCoder we will keep "is_valid" within the scope of  checking if entries populated by ROM are not invalid indexes or addresses (as per agreed markers) and add more extensive check in the mutable code.